docs/v1/P4Runtime-Spec.mdk

```diff
@@ -828,14 +829,10 @@ the server must return an `INVALID_ARGUMENT` error.
 
 ## Mastership Change { #sec-master-change}
 
-"Mastership change" refers to either one of these cases:
-
-1. A new `MasterArbitrationUpdate` is received from an already connected
-   controller for a given (`device_id`, `role_id`), which changes the controller
-   mastership status (the controller becomes master or slave).
-
-2. A streaming channel for a given master controller breaks, forcing a new
-   master to be elected.
+"Mastership change" refers to the case where a new `MasterArbitrationUpdate` is
```

Comment:
I feel we should disambiguate various cases that lead to mastership change:

1. Mastership is gained when a connected instance (other than the currently selected master) advertises a new higher election_id.
2. Mastership is updated when current master advertises a new highest election_id.
3. Mastership is lost when either (i) streaming channel for the current master breaks or (ii) current master advertises a new election_id lower than (but not used by) the election_ids advertised by other controller connections (for this role_id).

Open question: Should the protocol explicitly disallow 3(ii), i.e. require election_id to be monotonically increasing?

Reply:
I vote for disallowing 3(ii)  as it can produce corner-cases, e.g. the switch now considering master a client that does not expect to be master

Reply:
It seems fine to allow 3 (ii). e.g. if a master wants to downgrade itself to read only.

Reply:
3 is not a "mastership change", as no new master will be chosen.

Reply:
3i and ii are both allowed and explicitly documented in the updated PR.  @ccascone had concerns that the switch might start considering a client to be master that doesn't expect to be master, but that can't happen.  If a master downgrades (or loses connection), then the switch will be without master.  This is explicitly documented in the spec now.  Let me know if there are still concerns.